app/domain/authentication/authn_jwt/validate_status.rb

```diff
@@ -134,25 +144,15 @@ def webservice
       end
 
       def validate_signing_key
-        fetch_signing_key.call
-        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedSigningKeyConfiguration.new)
-      end
-
-      def fetch_signing_key
-        @fetch_signing_key ||= ::Util::ConcurrencyLimitedCache.new(
-          ::Util::RateLimitedCache.new(
-            fetch_signing_key_interface,
-            refreshes_per_interval: ::Authentication::AuthnJwt::CACHE_REFRESHES_PER_INTERVAL,
-            rate_limit_interval: ::Authentication::AuthnJwt::CACHE_RATE_LIMIT_INTERVAL,
-            logger: @logger
-          ),
-          max_concurrent_requests: ::Authentication::AuthnJwt::CACHE_MAX_CONCURRENT_REQUESTS,
-          logger: @logger
+        @fetch_signing_key.call(
+          cache_key: signing_key_interface.signing_key_uri,
```

Comment:
May be it's better to call `signing_key_interface` explicit once and then approach the member it creates